lib/lightning/credentials/credential.ex

```diff
@@ -50,33 +53,45 @@ defmodule Lightning.Credentials.Credential do
       message: "you have another credential with the same name"
     )
     |> assoc_constraint(:user)
-    |> assoc_constraint(:oauth_client)
     |> validate_format(:name, ~r/^[a-zA-Z0-9_\- ]*$/,
       message: "credential name has invalid format"
     )
-    |> validate_oauth()
+    |> maybe_validate_oauth_fields(attrs)
```

Comment:
I'm wondering here, could we not use `cast_assoc` for the oauth stuff, I see there are two places that call `Credentials.validate_oauth_token_data`; would it not be better to have the oauth body be validated by the OAuth token exclusively, since `Credential` won't have the body anymore - please correct me if my understanding is wrong.